src/api.jl

```diff
@@ -196,51 +408,78 @@ function infer(fieldids::Vector{Tuple{Int, Int, Int}},
     # initialize tiled images and model parameters for trimming.  We will
     # initialize the psf again before fitting, so we don't do it here.
     info("initializing celeste without PSF fit")
+    tic()
     tiled_images = SkyImages.break_blob_into_tiles(images, TILE_WIDTH)
     mp = ModelInit.initialize_model_params(tiled_images, images, catalog,
                                            fit_psf=false)
+    times.init_mp = toq()
 
     # get indicies of all sources relevant to those we're actually
     # interested in, and fit a local PSF for those sources (since we skipped
     # fitting the PSF for the whole catalog above)
     info("fitting PSF for all relevant sources")
+    tic()
     ModelInit.fit_object_psfs!(mp, target_sources, images)
+    times.fit_psf = toq()
+
+    # need a ModelParams per thread
+    mp_array = Array(ModelParams, nthreads())
+    mp_array[1] = mp
+    for i = 2:nthreads()
+        mp_array[i] = deepcopy(mp)
+    end
 
+    # iterate over sources
     results = Dict{Int, Dict}()
-    for s in target_sources
+    results_lock = SpinLock()
+    tic()
+    @threads for s in target_sources
+        tid = threadid()
         entry = catalog[s]
-        mp.active_sources = [s]
+        mp_array[tid].active_sources = [s]
 
         try
-            info("===================================================")
-            info("processing source $s: objid= $(entry.objid)")
+            nputs(dt_nodeid, "processing source $s: objid = $(entry.objid)")
+            tic()
 
             t0 = time()
-            relevant_sources = ModelInit.get_relevant_sources(mp, s)
-            mp_source = ModelParams(mp, relevant_sources)
+            relevant_sources = ModelInit.get_relevant_sources(mp_array[tid], s)
+            mp_source = ModelParams(mp_array[tid], relevant_sources)
```

Comment:
Then after this, you need to set mp_source.active_sources to [ findfirst(relevant_sources, s) ].  However, it occurs to me that maybe it'll be easier for me to make these changes after you merge, so feel free to leave it as mp_array for now and I'll make a PR to change this.  

It will work fine-as is, there's just an unnecessary copy.